tokio-timer/src/timer/entry.rs

```diff
@@ -0,0 +1,559 @@
+use Error;
+use atomic::AtomicU64;
+use timer::{Handle, Inner};
+
+use futures::Poll;
+use futures::task::AtomicTask;
+
+use std::cell::UnsafeCell;
+use std::ptr;
+use std::sync::{Arc, Weak};
+use std::sync::atomic::{AtomicBool, AtomicPtr};
+use std::sync::atomic::Ordering::SeqCst;
+use std::time::Instant;
+use std::u64;
+
+/// Internal state shared between a `Sleep` instance and the timer.
+///
+/// This struct is used as a node in two intrusive data structures:
+///
+/// * An atomic stack used to signal to the timer thread that the entry state
+///   has changed. The timer thread will observe the entry on this stack and
+///   perform any actions as necessary.
+///
+/// * A doubly linked list used **only** by the timer thread. Each slot in the
+///   timer wheel is a head pointer to the list of entries that must be
+///   processed during that timer tick.
+#[derive(Debug)]
+pub struct Entry {
```

Comment:
pub(crate) ?

Reply:
It is not re-exported, but it is probably safer to be explicit.